src/main/java/seedu/apollo/command/task/AddCommand.java

```diff
@@ -276,15 +278,27 @@ private void warnEventModuleClash(Ui ui, Calendar calendar, Event event) {
     private boolean isClashingEventModule(ArrayList<CalendarModule> calendarModule, LocalDateTime eventStart,
                                           LocalDateTime eventEnd, String currentDate) {
 
-        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd/MM/yyyy HHmm");
+        DateTimeFormatter formatterFull = DateTimeFormatter.ofPattern("dd/MM/yyyy HHmm");
+        DateTimeFormatter formatterPartial = DateTimeFormatter.ofPattern("dd/MM/yyyy");
         for (CalendarModule module : calendarModule) {
             if (module.getSchedule() == null) {
                 continue;
             }
+
+            LocalDate date = LocalDate.parse(currentDate, formatterPartial);
+
+            int currentWeek = getWeekNumber(date);
+
+            ArrayList<Integer> weeks = module.getSchedule().getWeeks();
+
+            if (!weeks.contains(currentWeek)) {
+                continue;
+            }
+
             String moduleStartString = currentDate + " " + module.getSchedule().getStartTime();
             String moduleEndString = currentDate + " " + module.getSchedule().getEndTime();
-            LocalDateTime lessonStart = LocalDateTime.parse(moduleStartString, formatter);
-            LocalDateTime lessonEnd = LocalDateTime.parse(moduleEndString, formatter);
+            LocalDateTime lessonStart = LocalDateTime.parse(moduleStartString, formatterFull);
+            LocalDateTime lessonEnd = LocalDateTime.parse(moduleEndString, formatterFull);
 
             if (isEventLessonClashing(eventStart, eventEnd, lessonStart, lessonEnd) &&
                     isDuringSemester(eventStart,eventEnd)) {
```

Comment:
I believe `isDuringSemester()` is no longer necessary because above already checks the week which is based on whether it is in the semester. So we can remove this method.